store/src/java/com/zimbra/cs/service/UserServlet.java

```diff
@@ -907,7 +907,7 @@ private void resolveFormatter(UserServletContext context) throws UserServletExce
         if (context.formatter == null) {
             context.formatter = FormatterFactory.mFormatters.get(context.format);
             // check if its native formatter, and if onlyoffice formatter is available
-            if (context.formatter.getType().equals(FormatType.HTML_CONVERTED)) {
+            if (context.formatter.getType().equals(FormatType.HTML_CONVERTED) && (context.getFmt() == null || !context.getFmt().equals("eml"))){
```

Comment:
why do we need to check specific format here?

Reply:
If we don't check the format, it will use OnlyOffice formatter which doesn't support eml files.